helper/helper/src/main/java/com/linkedin/avroutil1/compatibility/AvroRecordUtil.java

```diff
@@ -711,6 +711,11 @@ protected static Object deepConvert(
           outputMap.put(outputKey, outputElement);
         }
         return outputMap;
+
+      case UNION:
+        inputValueActualSchema = AvroSchemaUtil.resolveUnionBranchOf(inputValue, inputSchema);
```

Comment:
this switch is on the OUTPUT schema type. is it possible that in case of schema evolution the input schema is not a union?

Reply:
Possibly, if input and output schemas are different. Which would result in an exception.

Reply:
can we add a TODO here? or try this in a unit test?

Reply:
update: newBuilder(otherBuilder) -> deepCopy flow won't allow another schema version so this shouldn't be possible there is another flow which uses this?